xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -1233,21 +1233,88 @@ public String getRenderedContent(boolean transformationContextIsolated, XWikiCon
+     *
+     * @param context  the XWiki Context object
+     * @return  the rendered content of the document or its translation.
+     * @throws XWikiException in case of error during the rendering.
+     * @since 11.3RC1
+     */
+    @Unstable
+    public String displayDocument(XWikiContext context) throws XWikiException
+    {
+        return displayDocument(getOutputSyntax(),context);
+    }
+
+    /**
+     * Execute and render the current document in the current context.
+     * The code is executed with right of this document content author.
+     *
+     * @param targetSyntax  the syntax to use to render the document
+     * @param context  the XWiki Context object
+     * @return  the rendered content of the document or its translation.
+     * @throws XWikiException in case of error during the rendering.
+     * @since 11.3RC1
+     */
+    @Unstable
+    public String displayDocument(Syntax targetSyntax, XWikiContext context) throws XWikiException
+    {
+        return getRenderedContent(targetSyntax, true, context, false);
+    }
+
+    /**
+     * Execute and render the document or its translation in the current context.
+     * The code is executed with right of this document (or the translation) content author.
+     * The translations are retrieved if they exist and based on XWiki preferences
+     * (see {@link #getTranslatedDocument(XWikiContext)}).
+     *
+     * @param targetSyntax  the syntax to use to render the document
+     * @param transformationContextIsolated see {@link DocumentDisplayerParameters#isTransformationContextIsolated()}
+     * @param context  the XWiki Context object
+     * @return  the rendered content of the document or its translation.
+     * @throws XWikiException in case of error during the rendering.
+     */
     public String getRenderedContent(Syntax targetSyntax, boolean transformationContextIsolated, XWikiContext context)
         throws XWikiException
+    {
+        return getRenderedContent(targetSyntax, transformationContextIsolated, context, true);
+    }
+
+    /**
+     * Execute and render the document or its translation in the current context.
+     * The code is executed with right of this document (or the translation) content author.
+     *
+     * @param targetSyntax the syntax to use to render the document
+     * @param transformationContextIsolated see {@link DocumentDisplayerParameters#isTransformationContextIsolated()}
+     * @param context the XWiki Context object
+     * @param retrieveTranslation if true retrieve the translation of the document according to the preferences (see
+     *  {@link #getTranslatedDocument(XWikiContext)}). If false, render the current document.
+     * @return the rendered content of the document or its translation.
+     * @throws XWikiException in case of error during the rendering.
+     * @since 11.3RC1
+     */
+    public String getRenderedContent(Syntax targetSyntax, boolean transformationContextIsolated, XWikiContext context,
```

Comment:
I would keep this new method private as internal helper since that's not the one you are using from what I understand. Otherwise it's not clear why we need to introduce displayDocument since you could use this one.

Reply:
OK, makes sense.

Reply:
Done.